src/sqlfluff/cli/formatters.py

```diff
@@ -236,9 +236,9 @@ def _format_config(self, linter):
                 ("sqlfluff", get_package_version()),
                 ("python", get_python_version()),
                 ("implementation", get_python_implementation()),
-                ("dialect", linter.dialect.name),
+                ("dialect", linter.config.get("dialect_obj").name),
                 ("verbosity", self._verbosity),
-            ] + linter.templater.config_pairs()
+            ] + linter.config.get("templater_obj").config_pairs()
```

Comment:
This might be big enough to separate into its own PR and issue wdyt? (My concern is that the conversation on this specific matter delays changes that are unrelated)

Without this change, can this PR still be used to lint a multi-dbt-project repository?

Reply:
> This might be big enough to separate into its own PR and issue wdyt? (My concern is that the conversation on this specific matter delays changes that are unrelated)
> 
> Without this change, can this PR still be used to lint a multi-dbt-project repository?

@dmateusp Good point. I can simplify this PR so that it only covers the functionality to run sqlfluff from outside the dbt directory and leave the issue of changing/not changing templaters with nested .sqlfluff files for another issue.